ec/ecresource/deploymentresource/elasticsearch/v2/schema.go

```diff
@@ -483,12 +484,17 @@ func elasticsearchTopologySchema(options topologySchemaOptions) schema.Attribute
 		nodeRolesPlanModifiers = append(nodeRolesPlanModifiers, SetUnknownOnTopologySizeChange())
 	}
 
+	var topologyPlanModifiers []planmodifier.Object
+	if options.tierName == "master" {
+		topologyPlanModifiers = append(topologyPlanModifiers, objectplanmodifier.UseStateForUnknown())
```

Comment:
But it makes sense to test the migration change to be sure it works.

Reply:
> Hmm probably it doesn't make a difference 🤔 Technically this plan-modifier is not needed, as the other dedicated-master modifier always sets the tier either to a value or null.

I think the new dedicated master modifier can change plan only if `master` tier is not explicitly configured.

Reply:
Yeah but then the field is not unknown anymore and neither of these modifiers applies.

Reply:
Yeah, indeed.

Reply:
There are probably some corner-cases where the dedicated-master-modifier fails and doesn't apply anything. In those cases it would fall back to `UseStateForUnknown`.

Reply:
Actually in my testing I found that I was wrong: The `UseTopologyStateForUnknown` is needed on the master tier as well. Otherwise it will not properly keep the child fields `Unknown`. The reason is that every modifier does actually run, so `UseStateForUnknown` runs first, puts state everywhere. Then the children are not unknown anymore and their plan modifier does not run.

So you were totally right that we need `UseTopologyStateForUnknown` in this case as well.

Reply:
Honestly, I forgot about the fact that plan modifiers execute in a certain order which matters.